src/engine/currencyEngine.js

```diff
@@ -673,6 +692,8 @@ export class CurrencyEngine {
   saveTx (edgeTransaction: EdgeTransaction): Promise<void> {
     this.logEdgeTransaction(edgeTransaction, 'Saving')
     this.engineState.saveTx(edgeTransaction.txid, edgeTransaction.signedTx)
+    const bcoinTxUuid = edgeTransaction.otherParams.bcoinTxUuid
```

Comment:
Some places have a lot of validation that this id exists, and others less. Any reason?

Reply:
I added as much validation as the previous code had. So in sign, broadcast, save, I don't validate much at all. If the object is missing it would have been missing in prior code and thrown such that the app would have to catch it. I assume that in the logging code, @thehobbit85  didn't want to throw only because of logging code so he added extra validation.